### ISSUE TITLE
Is this project still alive?

### ISSUE THREAD

**Issue by PoignardAzur** [headcrab-rs/headcrab#132]
With a title like that, there's no need for much elaboration.

The project generated a fair amount of enthusiasm when it was announced, but now it seems somewhat abandoned:

- No progress report on the official blog for 6 months.
- Last commit was November 26 2020,  more than 3 months ago.

Is the project still alive? Do the contributors still believe that the long-term goals can be realistically achieved given their available time to work on it and existing corporate interest, if any?

**Comment by bjorn3** [comment]
You may want to read https://headcrab.zulipchat.com/#narrow/stream/248039-general/topic/still.20active.3F/near/225623269 from 9 Feb by @nbaksalyar. Quoting part of it:

> [...] lately I haven't been able to find much spare time as I would like to work on open source, but the design/architecture/planning phase is active. I'm sorry for a lack of updates and uncertainty - that's my fault. With regards to funding and OpenCollective contributions: the funds are not being used at this time, and I'm happy to do refunds. That said, all unused funds eventually will be directed towards further development (for issue bounties, contributor rewards, infrastructure, etc.).
>
> The thing I'm working on right now is a development plan for 2021: I believe the focus should be on making Headcrab a practical, usable toolkit first and foremost - I see multiple simple use cases where it can provide value. The actual coding might take a bit of time, but abandoning this project was never an intent - personally, I see it as a long-term one. :) [...]

**Comment by ozkriff** [comment]
any updates on the state of the project?

**Comment by romatthe** [comment]
Yeah, an update would be interesting. I don't need a refund or anything like that, but I may decide to put my donation money towards another project if activity is low at the moment. I totally understand the situation of course, I just don't have that much spare cash and I'm already donating to way too many projects as it is.

**Comment by nbaksalyar** [comment]
@romatthe, I'm preparing a more elaborate update about the current status, but a short version is that unfortunately there hasn't been much progress due to my personal circumstances. I plan for the development to kick in again starting this year.

Regarding the funding situation: I've contacted OpenCollective and unfortunately they don't have an option to process mass refunds automatically, so, in short, the options are:

1. Manual refunds. Just shoot me an email and I'll sort it quickly;
2. Funds can be redirected to other OSS project of your choice (you can also send me an email if that's your preference);
3. Your contributions will be used towards funding future bounties/issues and mini-projects within Headcrab.

All new donations should be paused at this time and the future funding model is going to be reconsidered.

And thank you very much for all your support! I'm sorry that the project hasn't been progressing as quickly as it was intended.

**Comment by theIDinside** [comment]
> @romatthe, I'm preparing a more elaborate update about the current status, but a short version is that unfortunately there hasn't been much progress due to my personal circumstances. I plan for the development to kick in again starting this year.
> 
> Regarding the funding situation: I've contacted OpenCollective and unfortunately they don't have an option to process mass refunds automatically, so, in short, the options are:
> 
>     1. Manual refunds. Just shoot me an email and I'll sort it quickly;
> 
>     2. Funds can be redirected to other OSS project of your choice (you can also send me an email if that's your preference);
> 
>     3. Your contributions will be used towards funding future bounties/issues and mini-projects within Headcrab.
> 
> 
> All new donations should be paused at this time and the future funding model is going to be reconsidered.
> 
> And thank you very much for all your support! I'm sorry that the project hasn't been progressing as quickly as it was intended.

I have quite a lot of spare time right now, as I'm in between / looking for jobs, and I would *love* to be able to put in some work into this, this seems like a match made in heaven for what my passions are wrt to software development, rust, low level, debugger stuff that I would like to develop personal experience & skills for. But right now, navigating the issues and feature requests is kind of a maze, and I would love if you could just clean up *just* a little bit / fire off some quick to do list or whatever, so that someone like me can get going on it!

**Comment by bjorn3** [comment]
I went through the issues. I think I closed all fixed issues. I agree the issue tracker was a mess. I unfortunately don't have a quick to do list for you as it has been ages since I last looked at the source code.

**Comment by pro465** [comment]
I'm sad that this seems abandoned 😭

**Comment by PoignardAzur** [comment]
Keep in mind there's a lot on everyone's plate.

Sometimes you have big ambitious project ideas, and when you start working on them you realize there's a lot of foundational work that needs to happens first. It was my case on other rust projects.

Just because there hasn't been recent progress doesn't mean the project is abandoned.

**Comment by nbaksalyar** [comment]
@PoignardAzur, thank you, this is indeed the case.

This is a large project that will take lots of time to mature but the current goal is to make it usable in actual, day-to-day development sooner rather than later (i.e., focusing more on small dev tools rather than the library and foundational stuff for the time being). So there's going to be more activity and updates.

**Comment by pro465** [comment]
@PoignardAzur @nbaksalyar 

so this is alive? yay!

anyways, just letting yall know that i might get involved once i have the resources. see yall till then, and thanks for raising my hopes up again!
